src/components/PaymentStripe/checkoutForm.js

```diff
@@ -0,0 +1,370 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+    useElements,
+    useStripe,
+    CardElement,
+    CardNumberElement,
+    CardExpiryElement,
+    CardCvcElement,
+} from '@stripe/react-stripe-js';
+import { useTheme } from '../../libs/hooks';
+import Input from '../Input';
+import Button from '../Button';
+import GoogleAddressLookup from '../GoogleAddressLookup';
+import RequiredAsterisk from '../RequiredAsterisk';
+import RenderIf from '../RenderIf';
+import Spinner from '../Spinner';
+import Container from './styled/container';
+import FlexContainer from './styled/flexContainer';
+import FlexItem from './styled/flexItem';
+import ButtonContainer from './styled/buttonContainer';
+import LabelText from './styled/labelText';
+import ErrorText from '../Input/styled/errorText';
+import IconContainer from './styled/iconContainer';
+import CalendarIcon from './icons/calendarIcon';
+import UserIcon from './icons/userIcon';
+import CardIcon from './icons/cardIcon';
+import LockIcon from './icons/lockIcon';
+
+const loadingStyle = { visibility: 'collapse' };
+const rememberMeStyle = { marginTop: '10px' };
+const searchOptions = { types: ['address'] };
+const initialBillingDetail = {
+    name: '',
+    email: '',
+    phone: '',
+    address: null,
+    zipCode: '',
+};
+
+const CheckoutForm = props => {
+    const {
+        apiKeyGMap,
+        buttonLabel,
+        buttonAlign,
+        inline,
+        hasName,
+        hasEmail,
+        hasPhone,
+        hasAddress,
+        hasRememberMe,
+        onChange,
+        onProcessed,
+        onError,
+        getPaymentIntent,
+    } = props;
+    const stripe = useStripe();
+    const elements = useElements();
+    const theme = useTheme().rainbow;
+    const [loading, setLoading] = useState(true);
+    const [billingDetails, setBillingDetails] = useState(initialBillingDetail);
+    const [cardDetails, setCardDetails] = useState({});
+    const [rememberMe, setRememberMe] = useState(false);
+    const [error, setError] = useState();
+    const cardElementOptions = {
```

Comment:
every time we create an object in a component that we know we can memoize I think we should because it will be created differently every time and it will be pass to a child component, then the framework it won't be able to do proper optimization because the component is receiving a different object